.gitlab-ci.yml

```diff
@@ -0,0 +1,164 @@
+      fi
+    - echo "========= Installing SSH keys ========"
+    - 'command -v ssh-agent >/dev/null || ( apt-get update -y && apt-get install openssh-client -y )'
+    - eval $(ssh-agent -s)
+    - chmod 400 $OREGON_CI_KEY
+    - ssh-add $OREGON_CI_KEY
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+
+    - echo "======= Download configuration ========"
+    - ./Tests/scripts/download_demisto_conf.sh
+
+    - echo "========== Build Parameters =========="
+    - set | grep -E "^NIGHTLY=|^INSTANCE_TESTS=|^SERVER_BRANCH_NAME=|^ARTIFACT_BUILD_NUM=|^DEMISTO_SDK_NIGHTLY=|^TIME_TO_LIVE=|^CONTRIB_BRANCH=|^FORCE_PACK_UPLOAD=|^PACKS_TO_UPLOAD=|^BUCKET_UPLOAD=|^GCS_MARKET_BUCKET=|^SLACK_CHANNEL="
+    - python --version
+    - python3 --version
+    - node --version
+    - npm --version
+    - demisto-sdk --version
+
+
+variables:
+  CONTENT_VERSION: "21.3.2"
+  SERVER_VERSION: "6.0.0"
+  GIT_SHA1: "6fee942518c6b2275a6cb61c1824574171cf58f9" # guardrails-disable-line disable-secrets-detection
+  DONT_CACHE_LAST_RESPONSE: "true"
+  GCS_MARKET_BUCKET: "marketplace-dist"
+  SLACK_CHANNEL: "dmst-content-team"
+  DEMISTO_README_VALIDATION: "true"
+  ARTIFACTS_FOLDER: "/builds/xsoar/content/artifacts"
+  BASH_ENV: "/builds/xsoar/content/artifacts/bash_env"
+  PYTHONPATH: "/builds/xsoar/content"
+  FEATURE_BRANCH_NAME: "v4.5.0"
+# Gitlab CI/CD uses shallow fetch with default depth of 50, these arguments are used to override it
+# See here for more about that https://docs.gitlab.com/ee/ci/pipelines/settings.html#git-shallow-clone  # disable-secrets-detection
+  GIT_STRATEGY: clone
+  GIT_DEPTH: 0
+
+
+stages:
+  - unittests-and-validations
+  - create-instances
+  - run-instances
+  - upload-to-marketplace
+
+create-instances:
+  stage: create-instances
+  script:
+    - |
+      demisto-sdk create-id-set -o ./Tests/id_set.json
+      cp ./Tests/id_set.json $ARTIFACTS_FOLDER
+    - python3 Utils/release_notes_generator.py $CONTENT_VERSION $CI_COMMIT_SHA $CI_BUILD_ID --output $ARTIFACTS_FOLDER/packs-release-notes.md --github-token $GITHUB_TOKEN
+    - ./Documentation/commonServerDocs.sh
+    -  demisto-sdk create-content-artifacts -a $ARTIFACTS_FOLDER --cpus 8 --content_version $CONTENT_VERSION
+    - |
+      gcloud auth activate-service-account --key-file="$GCS_ARTIFACTS_KEY" > auth.out 2>&1
+      echo "successfully activated google cloud service account"
+    - successful_feature_branch_build=$(gsutil ls "gs://xsoar-ci-artifacts/content/$FEATURE_BRANCH_NAME/*" | tail -n 1 | grep -o -E "content/$FEATURE_BRANCH_NAME/[0-9]*")
+    - echo $successful_feature_branch_build
+    - python3 Utils/merge_content_new_zip.py -f $FEATURE_BRANCH_NAME -b $successful_feature_branch_build
+    - zip -j $ARTIFACTS_FOLDER/uploadable_packs.zip $ARTIFACTS_FOLDER/uploadable_packs/* || ((($? > 0)) && echo "failed to zip the uploadable packs, ignoring the failure" && exit 0)
+    - rm -rf $ARTIFACTS_FOLDER/uploadable_packs
```

Comment:
why aren't those executed as the same step? they belong to the same logic unit

Reply:
Its more verbose to run commands one by one IMO.